### PR TITLE
Round to 5 decimal for embedding score to prevent flakiness

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: nl_server/search.py [hunk 1]
```diff
@@ -89,7 +89,7 @@ def _rank_vars(candidates: EmbeddingsResult,
         sv2sentences[dcid].add(c.sentence)
 
   for sv, score in sorted(sv2score.items(),
-                          key=lambda item: (-item[1], item[0])):
+                          key=lambda item: (-round(item[1], 6), item[0])):
```
Comment (pradh) [inline nl_server/search.py:92]: Maybe have TODO to truncate score using model parameter?

**Review by pradh — APPROVED** [review]
Nice!

File: nl_server/search.py (hunk 1 above)
Comment (pradh) [reply nl_server/search.py:92]: Oh also leave a comment why we do it

Comment (shifucun) [reply nl_server/search.py]: Done

File: nl_server/search.py [hunk 2]
```diff
@@ -88,8 +88,11 @@ def _rank_vars(candidates: EmbeddingsResult,
             dvars.SentenceScore(sentence=c.sentence, score=c.score))
         sv2sentences[dcid].add(c.sentence)
 
+  # TODO: truncate the score based on model parameters from yaml
+  # Same model would produce different scores after certain decimals, so we want
+  # to round to 6 decimal places to make the score and rank stable.
   for sv, score in sorted(sv2score.items(),
-                          key=lambda item: (-item[1], item[0])):
+                          key=lambda item: (-int(item[1] * 1e6), item[0])):
```
Comment (pradh) [inline nl_server/search.py:95]: curious why multiply and turn to int?

Comment (shifucun) [reply nl_server/search.py]: revert back to round. The tests were failing and I thought it's the float comparing issue. But looks like the original sort did not really take effect. Looks like if I create a new variable of the sorted result it works..

**Pull request merged**